wa/starter/src/main/java/org/apache/syncope/wa/starter/mapping/RegisteredServiceMapper.java

```diff
@@ -0,0 +1,161 @@
+
+    protected final Map<String, AuthMapper> registeredServiceAuthenticationPolicyMappers = new HashMap<>();
+
+    protected final Map<String, AccessMapper> accessPolicyConfMappers = new HashMap<>();
+
+    protected final Map<String, AccessMapper> registeredServiceAccessStrategyMappers = new HashMap<>();
+
+    protected final Map<String, AttrReleaseMapper> attrReleasePolicyConfMappers = new HashMap<>();
+
+    protected final Map<String, AttrReleaseMapper> registeredServiceAttributeReleasePolicyMappers = new HashMap<>();
+
+    protected final Map<String, ClientAppMapper> clientAppTOMappers = new HashMap<>();
+
+    protected final Map<String, ClientAppMapper> registeredServiceMappers = new HashMap<>();
+
+    @Override
+    public void setApplicationContext(final ApplicationContext ctx) throws BeansException {
+        this.ctx = ctx;
+    }
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        ctx.getBeansOfType(AuthMapper.class).forEach((name, bean) -> {
+            AuthMapFor authMapFor = ctx.findAnnotationOnBean(name, AuthMapFor.class);
+            if (authMapFor != null) {
+                authPolicyConfMappers.put(authMapFor.authPolicyConfClass().getName(), bean);
+                registeredServiceAuthenticationPolicyMappers.put(
+                        authMapFor.registeredServiceAuthenticationPolicyClass().getName(), bean);
+            }
+        });
+
+        ctx.getBeansOfType(AccessMapper.class).forEach((name, bean) -> {
+            AccessMapFor accessMapFor = ctx.findAnnotationOnBean(name, AccessMapFor.class);
+            if (accessMapFor != null) {
+                accessPolicyConfMappers.put(accessMapFor.accessPolicyConfClass().getName(), bean);
+                registeredServiceAccessStrategyMappers.put(
+                        accessMapFor.registeredServiceAccessStrategyClass().getName(), bean);
+            }
+        });
+
+        ctx.getBeansOfType(AttrReleaseMapper.class).forEach((name, bean) -> {
+            AttrReleaseMapFor attrReleaseMapFor = ctx.findAnnotationOnBean(name, AttrReleaseMapFor.class);
+            if (attrReleaseMapFor != null) {
+                attrReleasePolicyConfMappers.put(attrReleaseMapFor.attrReleasePolicyConfClass().getName(), bean);
+                registeredServiceAttributeReleasePolicyMappers.put(
+                        attrReleaseMapFor.registeredServiceAttributeReleasePolicyClass().getName(), bean);
+            }
+        });
+
+        ctx.getBeansOfType(ClientAppMapper.class).forEach((name, bean) -> {
+            ClientAppMapFor clientAppMapFor = ctx.findAnnotationOnBean(name, ClientAppMapFor.class);
+            if (clientAppMapFor != null) {
+                clientAppTOMappers.put(clientAppMapFor.clientAppClass().getName(), bean);
+                registeredServiceMappers.put(clientAppMapFor.registeredServiceClass().getName(), bean);
+            }
+        });
+    }
```

Comment:
I think it would be better if all of this was moved to the configuration class that creates the bean instance, removing the need to implement an interface and inject an application context. Injecting the application context is a discouraged practice, and should be avoided when possible. It also makes it more difficult to write unit tests for this.